src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeWriter.cs

```diff
@@ -651,5 +656,46 @@ public void Dispose()
             {
             }
         }
+
+        public struct StringLiteralWriter : IDisposable
```

Comment:
Oh, nonoo. Just make a helper method in `RuntimeBasicWriter`. You wont have to touch the CSharpCodeWriter at all.